plat/kvm/x86/lxboot.c

```diff
@@ -0,0 +1,160 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2022, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/essentials.h>
+#include <uk/arch/limits.h>
+#include <uk/arch/types.h>
+#include <uk/arch/paging.h>
+#include <uk/plat/bootstrap.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/plat/common/lcpu.h>
+#include <uk/plat/common/memory.h>
+#include <uk/plat/common/sections.h>
+#include <kvm-x86/lxboot.h>
+
+#define lxboot_crash(rc, msg, ...) ukplat_crash()
+
+void _ukplat_entry(struct lcpu *lcpu, struct ukplat_bootinfo *bi);
+
+static void
+lxboot_init_cmdline(struct ukplat_bootinfo *bi, struct lxboot_params *bp)
+{
+	struct ukplat_memregion_desc mrd = {0};
+	__u64 cmdline_addr;
+	__sz cmdline_size;
+	int rc;
+
+	cmdline_addr = bp->ext_cmd_line_ptr;
+	cmdline_addr <<= 32;
+	cmdline_addr |= bp->hdr.cmd_line_ptr;
+
+	cmdline_size = bp->hdr.cmd_line_size;
+
+	if (cmdline_addr == 0)
+		lxboot_crash(-EINVAL, "Command line address is zero");
+
+	if (cmdline_size == 0)
+		return;
+
+	mrd.pbase = cmdline_addr;
+	mrd.vbase = cmdline_addr;
+	mrd.len = cmdline_size;
+	mrd.type = UKPLAT_MEMRT_CMDLINE;
+	mrd.flags = UKPLAT_MEMRF_READ | UKPLAT_MEMRF_MAP;
+#ifdef CONFIG_UKPLAT_MEMRNAME
+	memcpy(mrd.name, "cmdline", sizeof("cmdline"));
+#endif
+
+	rc = ukplat_memregion_list_insert_nonempty(&bi->mrds, &mrd);
```

Comment:
Why use the `nonempty` variation here? Are you not already checking for the size above? Same for initrd.